### PR TITLE
Remove deprecated `StreamReader`

### LINKED ISSUE #1123: Useless TCP test
Looking at #1119 I noticed that tests use TCP with seemingly the only intention being testing whether the code can handle broken-up packets. This can be achieved without TCP and without artificial delay of 1s considerably slowing down the tests. I've seen an entire library to simulate breaks at random points, I just can't remember how it was called. But it can be done manually too:

```rust
// Breaks up reading to force multiple `read` calls when it reaches the end of each part
struct BreakingReader {
    // Changing to iterators left as an exercise for the reader
    data: Vec<Vec<u8>>,
    idx: usize,
    pos: usize,
}

impl BreakingReader {
    fn new(data: Vec<Vec<u8>>) -> Self {
        BreakingReader {
            data,
            idx: 0,
            pos: 0,
        }
    }
}

impl io::Read for BreakingReader {
    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
        if self.idx >= self.data.len() {
            return Ok(0)
        }
        // We never let pos to go out of bounds
        let part = &self.data[self.idx][self.pos..];
        let to_copy = part.len().min(buf.len());
        buf[..to_copy].copy_from_slice(&part[..to_copy]);
        if to_copy == part.len() {
            self.pos = 0;
            self.idx += 1;
        } else {
            self.pos += to_copy;
        }
        Ok(to_copy)
    }
}
```

Then just replace `serve_tcp()` with `BreakingReader::new()` and remove the `handle`/`join`.

### PR DESCRIPTION
**Pull request by Kixunil**
`StreamReader` was deprecated for a while, yet we needlessly maintain it
and it eats our testing time. This change removes it completely.

Closes #1123

### CODE REVIEW

File: src/blockdata/block.rs [hunk 1]
```diff
@@ -533,20 +533,18 @@ mod tests {
...
     pub fn bench_stream_reader(bh: &mut Bencher) {
         let big_block = include_bytes!("../../test_data/mainnet_block_000000000000000000000c835b2adcaedc20fdf6ee440009c249452c726dafae.raw");
         assert_eq!(big_block.len(), 1_381_836);
         let big_block = black_box(big_block);
 
         bh.iter(|| {
-            let mut reader = StreamReader::new(&big_block[..], None);
-            let block: Block = reader.read_next().unwrap();
+            let mut reader = &big_block[..];
+            let block = Block::consensus_decode(&mut reader).unwrap();
```
Comment (tcharding) [inline src/blockdata/block.rs:547]: I haven't groked the bench mark code in general but this function probably should be renamed to say what it is bench marking, possibly `bench_block_deserialize_logic`?

**Review by tcharding — APPROVED** [review]
ACK 0c9c14128b8225816a8fd2ebd4b1ca4b970605cf (with the function rename).

File: src/blockdata/block.rs (hunk 1 above)
Comment (Kixunil) [reply src/blockdata/block.rs:547]: Oh, I didn't read the function name. :sweat_smile: I will have to look if it's even needed, maybe it should be removed.

Comment (tcharding) [reply src/blockdata/block.rs:547]: Cheers, I could not work out if it was adding value or not.

**Review by sanket1729 — APPROVED** [review]
utACK 0c9c14128b8225816a8fd2ebd4b1ca4b970605cf.

**Review by apoelstra — APPROVED** [review]
ACK 0c9c14128b8225816a8fd2ebd4b1ca4b970605cf

File: src/blockdata/block.rs (hunk 1 above)
Comment (apoelstra) [reply src/blockdata/block.rs:547]: My intuition is we should probably drop it. Can do this in a followup PR

**Pull request merged**